tools/Performance/DynamoPerformanceTests/DynamoViewPerformanceTestBase.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using BenchmarkDotNet.Attributes;
+using DynamoCoreWpfTests;
+
+namespace DynamoPerformanceTests
+{
+
+    public class DynamoViewPerformanceTestBase : NodeViewTests
+    {
+        /// <summary>
+        /// Override this function to preload dlls into Dynamo library
+        /// </summary>
+        /// <param name="libraries">extra dlls to load</param>
+        protected override void GetLibrariesToPreload(List<string> libraries)
+        {
+            base.GetLibrariesToPreload(libraries);
+
+            libraries.Add("VMDataBridge.dll");
+            libraries.Add("ProtoGeometry.dll");
+            libraries.Add("DesignScriptBuiltin.dll");
+            libraries.Add("DSCoreNodes.dll");
+            libraries.Add("DSOffice.dll");
+            libraries.Add("DSIronPython.dll");
+            libraries.Add("FunctionObject.ds");
+            libraries.Add("BuiltIn.ds");
+            libraries.Add("DynamoConversions.dll");
+            libraries.Add("DynamoUnits.dll");
+            libraries.Add("Tessellation.dll");
+            libraries.Add("Analysis.dll");
+            libraries.Add("GeometryColor.dll");
+            libraries.Add("FFITarget.dll");
+        }
+
+        /// <summary>
+        /// Test directory containing the graph suite which the current performance test will be running against
+        /// </summary>
+        public static string testDirectory;
+
+        /// <summary>
+        /// Automated creation of performance test cases, one for each
+        /// parameter source. Notice this attribute must be public with
+        /// [ParamsSource] tag which will only thrown runtime error otherwise.
+        /// </summary>
+        [ParamsSource(nameof(PerformanceTestSource))]
+        public static string DynamoFilePath { get; set; }
+
+        /// <summary>
+        /// Populates the test cases based on DYN files in the performance tests folder.
+        /// Notice this function must be public as well because it is 
+        /// defined as the ParamsSource of DynamoFilePath property.
+        /// Console app will throw runtime error otherwise.
+        /// </summary>
+        /// <returns></returns>
+        public static IEnumerable<string> PerformanceTestSource()
```

Comment:
Can we move `GetLibrariesToPreload` and `PerformanceTestSource` into a common base class or helper class as these methods are repeated for both the model and view test classes.

Reply:
Since we already have a base class for test base here and C# does not support multiple class inheritance. I will try to see if I can move them to the helper

Reply:
@aparajit-pratap Please refer to https://benchmarkdotnet.org/articles/samples/IntroParamsSource.html. Param source tag can only apply to source within benchmark class, so this function has to stay.

Reply:
@aparajit-pratap Moved the common part of `GetLibrariesToPreload` to helper